source/dub/commandline.d

```diff
@@ -1809,6 +1813,31 @@ class UpgradeCommand : Command {
 		if (m_prerelease) options |= UpgradeOptions.preRelease;
 		if (m_dryRun) options |= UpgradeOptions.dryRun;
 		dub.upgrade(options, free_args);
+
+		if (m_includeSubPackages) {
+			bool any_error = false;
+
+			auto spacks = dub.project.rootPackage
+				.subPackages
+				.filter!(sp => sp.path.length);
+
+			foreach (sp; spacks) {
+				try {
+					auto fullpath = (dub.projectPath ~ sp.path).toNativeString();
+					logInfo("Upgrading sub package in %s", fullpath);
+					auto sdub = new Dub(fullpath, dub.packageSuppliers, SkipPackageSuppliers.all);
```

Comment:
why do we need to construct an extra dub instance? If it's not possible to easily make it work with the regular upgrade function a comment about this would be appreciated here for future readers.